packages/analytics-nextjs/src/components/Analytics.tsx

```diff
@@ -33,62 +22,13 @@ export function Analytics() {
         }
     }, [currentPath, onPageView, page, previousPath]);
 
-    /**
-     * @deprecated Improved campaign click tracking supersedes this functionality. To be removed in v2.0
-     */
-    useEffect(() => {
-        const userId = userRef.current().id();
-        const utm = getUrlParameters('utm_');
-        const id = utm.get('id');
-        const source = utm.get('source');
-        const medium = utm.get('medium');
-
-        if (id && source === 'email' && medium === 'campaign') {
-            getRecipientInfo(id)
-                .then((data) => {
-                    // re-map current user to the correct identifier
-                    if (userRef.current().id() === data.recipient_id) {
-                        aliasRef.current(data.id, id);
-                    }
-
-                    identifyRef.current(data.id);
-                    trackRef.current(CAMPAIGN.CLICK, { recipient_id: data.recipient_id }, () =>
-                        stripUrlParameters('utm_'),
-                    );
```

Comment:
@marko-ogg Since I've reworked the events, I had to drop the campaign tracking code here due to the events no longer existing, but I think that should be fine since the comments also mention that we're now tracking things in Sendgrid.

We've also been tracking asset clicks in stories in campaigns, but I believe all of these are handled by Sendgrid now (i.e. when I include a full story in campaign and I click on an image in the email).

Reply:
So does this mean that I should drop legacy tracking completely from backend?

We still have it in code, but it's not used. It was mostly because of the transition period between old way of tracking and Sendgrid.

Reply:
I'm not sure which legacy tracking you're referring to 🤔 These tracking calls were going to Plausible and Segment in the new themes, but I don't think we've actually used/shown them anywhere.

Reply:
Or maybe it was somehow synced from Segment to the app before the Sendgrid tracking rework? I'm not sure.

Reply:
This is how we were tracking clicks before, right? We were using UTM params before and pushing to Prezly analytics if I'm right.

So if this was the code that was working with old UTM params, it means I can also drop the support for legacy tracking on the backend.